galley/pkg/config/processing/runtime.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processing
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"istio.io/api/mesh/v1alpha1"
+
+	"istio.io/istio/galley/pkg/config/event"
+)
+
+// ProcessorOptions are options that are passed to event.Processors during startup.
+type ProcessorOptions struct {
+	MeshConfig   *v1alpha1.MeshConfig
+	DomainSuffix string
+}
+
+// RuntimeOptions is options for Runtime
+type RuntimeOptions struct {
+	Sources      []event.Source
```

Comment:
I think we should make this a single `event.Source` and provide a `CompositeSource` implementation.

Reply:
I'm not sure I see the concern. Either way would be fine... either the composite would pass in the handler directly to all children (i.e. no extra method call during event processing), or it would give them its own handler and then call the target handler (i.e. one extra method call during event processing). Either way there's likely to be no noticeable performance penalty at runtime, but it would simplify these constructs a bit.